nodestream/pipeline/extractors/stores/aws/dynamodb_extractor.py

```diff
@@ -0,0 +1,199 @@
+from logging import getLogger
+from typing import Any, Dict, List, AsyncGenerator
+
+from ...credential_utils import AwsClientFactory
+from ...extractor import Extractor
+
+
+class DynamoObject:
+    """
+        Dynamo objects generally come in the form {"attribute_type": value}
+        Example:
+            {"S", "some_string"}
+            {"N", "12345"}
+    """
+    def __init__(self, object: Dict[str, Any]):
+        self.attribute_type = list(object.keys())[0]
+        self.value = list(object.values())[0]
+
+    def items(self) -> tuple[str, Any]:
+        return self.attribute_type, self.value
+    
+    def as_dynamo_representation(self):
+        if isinstance(self.value, dict):
+            return {self.attribute_type: {key: value.as_dynamo_representation() for key, value in self.value.items()}}
+        elif isinstance(self.value, list):
+            return {self.attribute_type: [value.as_dynamo_representation() for value in self.value ]}
+        return {self.attribute_type: self.value}
+
+def leave_untouched(value):
+    return value
+
+def convert_scalar(value, type_handler):
+    return  None if type_handler is None else type_handler(value)
+
+def convert_array(array: List[Any], type_handler):
+    return [type_handler(value) for value in array]
+
+def convert_map(mapping: Dict[str, Any], type_handler):
+    return {key: type_handler(value) for key, value in mapping.items()}
+
+def convert_number(value: str):
+    if '.' in value:
+        return float(value)
+    return int(value)
+
+DEFAULT_HANDLER = (convert_scalar, leave_untouched)
+
+def convert_value(dynamo_object: Dict[str, Any]):
+    attribute_type, value = DynamoObject(dynamo_object).items()
+    result_handler, type_handler = ATTRIBUTE_TYPE_DESCRIPTOR_HANDLERS.get(attribute_type, DEFAULT_HANDLER)
+    return result_handler(value, type_handler)
```

Comment:
At first, I was going to complain that this function name isn't particularly communicative. A convert function probably needs to describe the `from` -> `to` of the conversion to be meaningful. 

Then I was looking at it. In reality, this thing has [Inappropriate Intimacy ](https://refactoring.guru/smells/inappropriate-intimacy) with `DynamoObject`. Maybe make this behavior a method on `DynamoObject` called something like `to_python_type` or the like.